kikuchipy/signals/ebsd.py

```diff
@@ -1004,6 +1005,275 @@ def dictionary_indexing(
             get_orientation_similarity_map=get_orientation_similarity_map,
         )
 
+    def orientation_refinement(
+        self,
+        xmap: CrystalMap,
+        master_pattern,
+        detector: EBSDDetector,
+        energy: Union[int, float],
+        mask: Optional[np.ndarray] = None,
+        method: Optional[str] = None,
+        method_kwargs: Optional[dict] = None,
+        trust_region: Optional[list] = None,
+        compute: bool = True,
+    ) -> CrystalMap:
+        """Performs an orientation refinement using the initial indexing
+        results stored in a single phase
+        :class:`~orix.crystal_map.CrystalMap` and the fixed
+        detector-sample geometry. The refinement uses a subset of the
+        optimization routines implemented in scipy and attempts to
+        maximize the similarity between experimental and simulated
+        patterns as calculated by the NCC.
+
+        Parameters
+        ----------
+        xmap : CrystalMap
+            A crystal map storing the results of the initial EBSD
+            indexing.
+        master_pattern : EBSDMasterPattern
+            EBSDMasterPattern in the square Lambert projection.
+        detector : EBSDDetector
+            EBSD detector describing the detector dimensions and the
+            detector-sample geometry with either a single, fixed
+            projection/pattern center or a projection center for each
+            scan point.
+        energy : int
+            Acceleration voltage, in kV, used to simulate the desired
+            master pattern.
+        mask : np.ndarray
+            Boolean mask to be applied to the simulated patterns.
+        method : str, optional
+            Name of the scipy.optimize function to be used. Must be one
+            of "minimize", "differential_evolution", "dual_annealing",
+            or "basinhopping". If not specified, "minimize" is used.
+        method_kwargs : dict, optional
+            Keyword arguments passed to the scipy.optimize function
+            specified above.
+        trust_region : list, optional
+            List of how wide the bounds, centered on the initial
+            orientation indexing result, should be for
+            (phi1, Phi, phi2) in degrees. Only used for methods that
+            support bounds (excluding Powell). Defaults to [1, 1, 1].
+        compute : bool
```

Comment:
What is it that must be called compute on if compute=False? This should be stated.